internal/docstore/docstore.go

```diff
@@ -50,6 +50,15 @@ func newCollection(d driver.Collection) *Collection {
 	return &Collection{driver: d}
 }
 
+// RevisionField is the name of the document field used for document revision
```

Comment:
Don't some providers have explicit support for this? What's the story for those that do?

Reply:
Many do. Drivers are obligated to follow these semantics, as verified in the new conformance tests in this PR. If you look at the changes to `fs.go` you'll see what that looks like in general: copying the native revision field into the returned document in a `Get`, and enforcing the necessary preconditions on writes.